staging/src/k8s.io/apiserver/pkg/authorization/cel/mapper.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	authorizationv1 "k8s.io/api/authorization/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type CELMapper struct {
+	CompilationResults []CompilationResult
+}
+
+// eval evaluates the given SubjectAccessReview cel mapping expression and returns a EvaluationResult.
+func (c *CELMapper) Eval(ctx context.Context, r *authorizationv1.SubjectAccessReview) ([]EvaluationResult, error) {
+	evaluations := make([]EvaluationResult, len(c.CompilationResults))
+	specVal, err := convertObjectToUnstructured(&r.Spec)
+	if err != nil {
+		return nil, fmt.Errorf("cel evaluation error: convert SubjectAccessReviewSpec object to unstructured failed in error: %w", err)
+	}
+	va := map[string]interface{}{
+		"request": specVal.Object,
+	}
+	for i, compilationResult := range c.CompilationResults {
+		var evaluation = &evaluations[i]
+		evalResult, _, err := compilationResult.Program.ContextEval(ctx, va)
+		if err != nil {
+			return nil, fmt.Errorf("cel evaluation error: expression '%v' resulted in error: %w", compilationResult.ExpressionAccessor.GetExpression(), err)
+		}
+		evaluation.EvalResult = evalResult
+	}
+	return evaluations, nil
+}
+
+func convertObjectToUnstructured(obj interface{}) (*unstructured.Unstructured, error) {
+	if obj == nil || reflect.ValueOf(obj).IsNil() {
+		return &unstructured.Unstructured{Object: nil}, nil
+	}
+	ret, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return nil, err
+	}
+	return &unstructured.Unstructured{Object: ret}, nil
```

Comment:
no need to wrap in unstructured.Unstructured if we just unwrap it when we use it at `"request": specVal.Object,`